src/System.Private.Interop/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -157,14 +157,15 @@ private static int SizeOfHelper(Type t, bool throwIfNotMarshalable)
                 return size;
             }
 
-            if (!typeHandle.IsBlittable() && !typeHandle.IsValueType())
-            {
-                throw new MissingInteropDataException(SR.StructMarshalling_MissingInteropData, t);
-            }
-            else
+            // IsBlittable() checks whether the type contains GC references. It is approximate check with false positives.
+            // This fallback path will return incorrect answer for types that do not contain GC references, but that are 
+            // not actually blittable; e.g. for types with bool fields.
```

Comment:
It would be nice to delete the `IsBlittable` check from here to avoid this problem at some point. I am not sure how much .NET Native depends on this fallback path, and what kind of impact it would have.

Reply:
Currently DynamicInterop struct marshalling depends on this fallback path.  These code still stays in .Net Native. mabye it is a good idea to move these code into CoreRT repro